runtime/util/utf8hash.c

```diff
@@ -23,15 +23,14 @@
 #include "util_api.h"
 
 /**
- * Hash function for UTF8 strings which works from the front of the string to the back.  Mathematically equivalent to
- * computeJavaHashForExpandedString, which works back to front on Unicode.
+ * Computes the hash value for an input string using the hash algorithm defined by the java/lang/String.hashCode()I
+ * method.
  *
- * @param data points to raw UTF8 bytes, assumed to be a valid encoding
- * @param length is the number of bytes
+ * @param data points to raw UTF8 bytes, assumed to be a valid (potentially multi-byte) encoding
+ * @param length the number of bytes to hash
  *
  * @return hash code for the UTF8 string
  */
-
 UDATA
 computeHashForUTF8(const U_8 * data, UDATA length)
```

Comment:
Does anyone still use this directly?  Can it be moved out of util and the c api be provided by calling the new impl in VMHelpers?

Reply:
> Does anyone still use this directly? Can it be moved out of util and the c api be provided by calling the new impl in VMHelpers?

I don't think so. There are several uses of the C API from within C contexts:

https://github.com/eclipse/openj9/blob/58a679f1fe4e1b7eaa97d2b01b1d9452df4edac0/runtime/shared_util/classpathcache.c#L267
https://github.com/eclipse/openj9/blob/58a679f1fe4e1b7eaa97d2b01b1d9452df4edac0/runtime/vm/classsupport.c#L1289
https://github.com/eclipse/openj9/blob/58a679f1fe4e1b7eaa97d2b01b1d9452df4edac0/runtime/vm/lockwordconfig.c#L57